apps/explorer/src/ui/TransactionCard.tsx

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ChevronRight12, ChevronRight16 } from '@mysten/icons';
+import clsx from 'clsx';
+import { type ReactNode, useCallback, useEffect, useState } from 'react';
+
+import { Card, type CardProps } from '~/ui/Card';
+import { Divider } from '~/ui/Divider';
+import { Heading } from '~/ui/Heading';
+import { Text } from '~/ui/Text';
+
+type Size = 'md' | 'sm';
+
+export function useTransactionCardControl(
```

Comment:
This feels like state that should be handled at the component level, not hoisted to the parent.

Reply:
Updated, this was done to handle the previous `show all` behavior